src/scene/vcVerticalMeasureTool.cpp

```diff
@@ -104,14 +108,22 @@ void vcVerticalMeasureTool::AddToScene(vcState *pProgramState, vcRenderData *pRe
       pInstance->renderFlags = vcRenderPolyInstance::RenderFlags_Transparent;
     }
 
-    m_labelInfo.worldPosition = m_points[1];
+    ResetLabelText();
+    m_labelList[0].worldPosition = (m_points[0] + m_points[1]) / 2;
+    m_labelList[1].worldPosition = (m_points[1] + m_points[2]) / 2;
+
     char labelBuf[128] = {};
     udStrcat(labelBuf, m_pNode->pName);
-    udStrcat(labelBuf, udTempStr("\n%s: %.3f", vcString::Get("sceneVerticalDistance"), udAbs(m_points[0].z - m_points[2].z)));
-    if(m_labelInfo.pText)
-      udFree(m_labelInfo.pText);
-    m_labelInfo.pText = udStrdup(labelBuf);
-    pRenderData->labels.PushBack(&m_labelInfo);
+    udStrcat(labelBuf, udTempStr("\n%s: %.3f", vcString::Get("sceneStraightDistance"), udMag3(m_points[0] - m_points[2])));
+    udStrcat(labelBuf, udTempStr("\n%s: %.3f", vcString::Get("sceneHorizontalDistance"), udMag2(m_points[0] - m_points[2])));
+    m_labelList[0].pText = udStrdup(labelBuf);
+
+    char labelBufVertical[128] = {};
+    udStrcat(labelBufVertical, udTempStr("%s: %.3f", vcString::Get("sceneVerticalDistance"), udAbs(m_points[0].z - m_points[2].z)));
```

Comment:
You could do this with `udSprintf` instead of both `udStrcat` and `udTempStr`.

Reply:
Fixed.